managed/resource_db_query_logging.go

```diff
@@ -0,0 +1,415 @@
+	state.GetAttribute(ctx, path.Root("account_id"), &config.AccountID)
+	state.GetAttribute(ctx, path.Root("project_id"), &config.ProjectID)
+	state.GetAttribute(ctx, path.Root("cluster_name"), &config.ClusterName)
+	state.GetAttribute(ctx, path.Root("cluster_id"), &config.ClusterID)
+	state.GetAttribute(ctx, path.Root("integration_name"), &config.IntegrationName)
+	state.GetAttribute(ctx, path.Root("config_id"), &config.ConfigID)
+}
+
+func (r resourceDbQueryLogging) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	var state DbQueryLoggingConfig
+
+	getConfigFromState(ctx, req.State, &state)
+	apiClient := r.p.client
+	accountId := state.AccountID.Value
+	projectId := state.ProjectID.Value
+	clusterId := state.ClusterID.Value
+	clusterName := state.ClusterName.Value
+	integrationName := state.IntegrationName.Value
+
+	dbqlConfig, readOK, message := resourceRead(ctx, accountId, projectId, clusterId, clusterName, integrationName, apiClient)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of Db Query log configuration associated with the cluster", message)
+		return
+	}
+
+	diags := resp.State.Set(ctx, &dbqlConfig)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r resourceDbQueryLogging) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	resp.Diagnostics.AddError(
+		"Unsupported Operation",
+		"Update is not currently supported.",
+	)
+}
+
+func (r resourceDbQueryLogging) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
+	resp.Diagnostics.AddError(
+		"Unsupported Operation",
+		"Delete is not currently supported.",
+	)
+}
+
+func (r resourceDbQueryLogging) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+	resp.Diagnostics.AddError("Import is not currently supported", "")
+
+}
+
+// Create a new Db Query Log Configuration for a Cluster
+func (r resourceDbQueryLogging) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var config DbQueryLoggingConfig
+	resp.Diagnostics.Append(getConfigFromPlan(ctx, req.Plan, &config)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the DB query log configuration")
+		return
+	}
+
+	apiClient := r.p.client
+	var accountId, projectId string
+	accountId, ok, message := getAccountId(ctx, apiClient)
+	if !ok {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, ok, message = getProjectId(ctx, apiClient, accountId)
```

Comment:
account id and project id have 1 - 1 mapping, if account id exists, then project id will be sure to exist. This check is unnecessary.

Reply:
@arishta-yb  it's possible for a project to be deleted while the account remains active. Deleting the last admin user in an account will remove the project, though the account itself persists. This can lead to a scenario where the account exists without an associated project, though it's relatively rare.

Reply:
As per `getProjectId` methods code(https://github.com/yugabyte/terraform-provider-ybm/blob/main/managed/common.go#L21), there are other errors that can occur(API call failure, multiple projects associated to an account).

Hence the error checking. Please let me know if I am misinterpreting.